File: AI-Outbound-BE-main/models/campaign.py
from pydantic import BaseModel
from typing import Optional, List

class Campaign(BaseModel):
    id: Optional[str] = None
    campaignName: str
    description: Optional[str] = None
    owner_id: str  # The user who created the campaign
    users: Optional[str] = None
    businessName: Optional[str] = None
    fullname: Optional[str] = None
    email: Optional[str] = None
    has_ebook: Optional[bool] = False
    ebook_path: Optional[str] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None

class CampaignCreate(BaseModel):
    campaignName: str
    description: Optional[str] = None
    users: Optional[str] = None
    campaignDate: Optional[str] = None
    fullname: Optional[str] = None
    email: Optional[str] = None
    has_ebook: Optional[bool] = False
    ebook_path: Optional[str] = None
    campaignTime: Optional[str] = None
class CampaignUpdate(BaseModel):
    campaignName: Optional[str] = None
    campaignDate: Optional[str] = None
    campaignTime: Optional[str] = None
    maxRetry: Optional[int] = None

File: AI-Outbound-FE-main/src/types/campaign.ts
export interface Campaign {
    id?: string;
    campaignName: string;
    description?: string;
    owner_id?: string;
    users?: string[];
    businessName?: string;
    fullname?: string;
    email?: string;
    hasEbook?: boolean;
    ebookPath?: string;
    created_at?: string;
    updated_at?: string;
    campaignDate?: string;
    maxRetry?: number;
    owner_name?: string;
    owner_email?: string;
}

export interface CampaignUser {
    id: string;
    name: string;
    email: string;
    role: string;
}
